meerk40t/core/elements.py

```diff
@@ -3258,21 +3272,562 @@ def element_grid(
+            # )
+
+            pts = []
+            myangle = startangle.as_radians
+            deltaangle = 2 * pi / corners
+            ct = 0
+            for j in range(corners):
+                if ct < alternate_seq:
+                    # print("Outer: Ct=%d, Radius=%.2f, Angle=%.2f" % (ct, radius, 180 * myangle / pi) )
+                    thisx = x_pos + radius * cos(myangle)
+                    thisy = y_pos + radius * sin(myangle)
+                else:
+                    # print("Inner: Ct=%d, Radius=%.2f, Angle=%.2f" % (ct, radius_inner, 180 * myangle / pi) )
+                    thisx = x_pos + radius_inner * cos(myangle)
+                    thisy = y_pos + radius_inner * sin(myangle)
+                ct += 1
+                if ct >= 2 * alternate_seq:
+                    ct = 0
+                if j == 0:
+                    firstx = thisx
+                    firsty = thisy
+                myangle += deltaangle
+                pts += [(thisx, thisy)]
+            # Close the path
+            pts += [(firstx, firsty)]
+            poly_path = Polygon(pts)
+            self.add_element(poly_path)
+            if data is None:
+                return "elements", [poly_path]
+            else:
+                data.append(poly_path)
+                return "elements", data
+
+        @context.console_argument(
+            "corners", type=int, help=_("Number of corners/vertices")
+        )
+        @context.console_argument(
+            "density", type=int, help=_("Amount of vertices to skip")
+        )
+        @context.console_argument(
+            "x_pos", type=Length, help=_("X-Value of polygon's center")
+        )
+        @context.console_argument(
+            "y_pos", type=Length, help=_("Y-Value of polygon's center")
+        )
+        @context.console_argument("radius", type=Length, help=_("Radius"))
+        @context.console_option(
+            "startangle", "s", type=Angle.parse, help=_("Start-Angle")
+        )
+        @context.console_option(
+            "inscribed",
+            "i",
+            type=bool,
+            action="store_true",
+            help=_("Shall the polygon touch the inscribing circle?"),
+        )
+        @context.console_command(
+            "star",
+            help=_(
+                "star <corners> <density> <x> <y> <r> <startangle> <inscribed> or star <corners> <density> <r>"
+            ),
+            input_type=("elements", None),
+            output_type="elements",
+        )
+        def element_star(
+            command,
+            channel,
+            _,
+            corners,
+            density,
+            x_pos,
+            y_pos,
+            radius,
+            startangle=None,
+            inscribed=None,
+            data=None,
+            **kwargs,
+        ):
+
+            if corners is None:
+                raise SyntaxError
+            if corners < 5:
+                raise SyntaxError(_("A star needs to have at least 5 corners"))
```

Comment:
Consider permitting `star 3 1`  shapes. While these are generally just going to be polyshapes. I see a lot of overlap between these commands.  It might be correct to combine these into just a `shape` command to make a wide variety of regular shapes. Where your needed values are `x,y` location and `points` and `radius` maybe permitting a wider sets of values since many of the other properties would be optional. Currently the `star` command cannot create isotoxal stars. The `shape`  is about as generic as things can get. And you could probably let people specify weird properties that tend to permit a lot of different geometry.

Reply:
If you decided to go with `shape` as a command. You might actually want to reduce corners to permit omission and default to 1. While this is actually just a dot. It's actually the case that making dots is functionally useful and we sort of need a method to do that since we actually have code to go to the location of a dot and fire the laser for a set period of time.